src/index.js

```diff
@@ -78,14 +79,33 @@ class HMIApp extends React.Component {
                 {
                     this.props.webEngineApps.map((app) => {
                         let query = `?sdl-host=${flags.CoreHost}&sdl-port=${flags.CoreWebEngineAppPort}&sdl-transport-role=${app.transportType.toLowerCase()}-server`;
-                        return (<WebEngineAppContainer key={app.policyAppID} policyAppID={app.policyAppID} iframeUrl={app.appUrl + app.entrypoint + query} />);
+                        var style = { display: 'none' };
+                        if (this.props.showWebView && app.runningAppId === this.props.activeAppId) {
+                            style = app.style;
+                        }
+                        return (<WebEngineAppContainer key={app.policyAppID}
+                            style={style}
+                            policyAppID={app.policyAppID}
+                            iframeUrl={app.appUrl + app.entrypoint + query} />);
                     })
                 }
             </div>
         )
     }
     componentDidMount() {
         this.sdl.connectToSDL()
+        if (window.performance.memory) { // chromium --enable-precise-memory-info
+            this.memoryUsageInterval = setInterval(function(self) {
+                var mem = window.performance.memory;
+                if (mem.usedJSHeapSize / mem.jsHeapSizeLimit > 0.75) {
+                    for (var app of self.props.webEngineApps) {
+                        if (app.runningAppId && app.style.position === 'absolute') {
+                            bcController.onExitApplication('RESOURCE_CONSTRAINT', app.runningAppId);
```

Comment:
If the app is taking up too much memory, should the HMI also "delete" the iframe? Does that happen indirectly after the OnExitApplication message is sent?

Reply:
This will indirectly remove the iframe after OnAppUnregistered is received by the HMI.